server/server.go

```diff
@@ -432,8 +423,8 @@ func (j *jsonRPCHub) GetPeers() int {
 	return len(j.Server.Peers())
 }
 
-func (j *jsonRPCHub) getAccountImpl(root types.Hash, addr types.Address) (*state.Account, error) {
-	snap, err := j.state.NewSnapshotAt(root)
+func getAccountImpl(state state.State, root types.Hash, addr types.Address) (*state.Account, error) {
+	snap, err := state.NewSnapshotAt(root)
 	if err != nil {
 		return nil, err
```

Comment:
Just a minor suggestion if it makes sense to you (inspiration found in `GetBalance` function which formatted error message in a similar manner).

```suggestion
		return nil, fmt.Errorf("unable to get snapshot for root '%s': %w", root, err)
```